bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosAlarm.java

```diff
@@ -12,29 +12,59 @@
  */
 package org.openhab.binding.sonos.internal;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * The {@link SonosAlarm} is a datastructure to describe
  * alarms in the Sonos ecosystem
  *
  * @author Karel Goderis - Initial contribution
  */
+@NonNullByDefault
 public class SonosAlarm implements Cloneable {
 
+    private final int id;
+    private String startTime;
+    private final String duration;
+    private final String recurrence;
+    private boolean enabled;
+    private final String roomUUID;
+    private final String programURI;
+    private final String programMetaData;
+    private final String playMode;
+    private final int volume;
+    private final boolean includeLinkedZones;
+
+    public SonosAlarm(int id, String startTime, String duration, String recurrence, boolean enabled, String roomUUID,
+            String programURI, String programMetaData, String playMode, int volume, boolean includeLinkedZones) {
+        this.id = id;
+        this.startTime = startTime;
+        this.duration = duration;
+        this.recurrence = recurrence;
+        this.enabled = enabled;
+        this.roomUUID = roomUUID;
+        this.programURI = programURI;
+        this.programMetaData = programMetaData;
+        this.playMode = playMode;
+        this.volume = volume;
+        this.includeLinkedZones = includeLinkedZones;
+    }
+
     @Override
-    public Object clone() {
+    public @Nullable Object clone() {
         try {
             return super.clone();
         } catch (Exception e) {
             return null;
         }
     }
```

Comment:
I don't think `clone` is ever supposed to return null normally. Honestly the use of `clone` is outdated to begin with and would much rather a custom method created for this instead. But if you don't want to go through the work of replacing the `clone` method I suggest you just throw some type of runtime exception instead of returning null. Also, you can override the return type as well.
```suggestion
    public SonosAlarm clone() {
        try {
            return (SonosAlarm) super.clone();
        } catch (Exception e) {
            throw new UnsupportedOperationException(e);
        }
    }
```

Reply:
I will try to suppress the clone method. 
We have another one in another class.

Reply:
I found no reason to implement Clonable in the class SonosAlarm !
Removed.